operator/controllers/kyma_controller.go

```diff
@@ -115,7 +119,7 @@ func (r *KymaReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
 	}
 
 	// create a remote synchronization context, and update the remote kyma with the state of the control plane
-	if kyma.Spec.Sync.Enabled {
+	if kyma.Status.State != v1alpha1.KymaStateDeleting && kyma.Spec.Sync.Enabled {
```

Comment:
make sense, I think this `Status.State != v1alpha1.KymaStateDeleting` better put into `replaceWithVirtualKyma` as a precondition to prevent update happens when kyma in deleting state.